api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java

```diff
@@ -26,5 +33,61 @@
 
   List<ReportingUser> getUsers();
 
-  List<ReportingWorkspace> getWorkspaces();
+  List<ReportingWorkspace> getWorkspaces(long limit, long offset);
+
+  default List<ReportingWorkspace> getWorkspaces() {
```

Comment:
The upshot of all these default methods is that we only need the iterator cruft in one place. It was fun to code, so hopefully not to painful to read.